bots/tests-data

```diff
@@ -0,0 +1,254 @@
+
+    # Ignore these sink urls. See below. This is an optimization.
+    # We don't need to retrieve the sink /status page for the head revision
+    ignore = { }
+
+    # Seed the set of sinks. We use these sinks to figure out additional
+    # revisions for the pull request. Unfortunately GitHub doesn't help us
+    # with a list of revisions that this pull request used to reflect. So
+    # we have to look to our sink for that info.
+    data = task.api.get("commits/{0}/status?page=1&per_page=100".format(head))
+    for status in data.get("statuses", [ ]):
+        url = status["target_url"]
+        if url:
+            ignore[urlparse.urljoin(url, "./")] = True
+            sink = urlparse.urljoin(url, "../")
+            if sink not in SINKS:
+                SINKS[sink] = links(sink)
+
+    # Now ask each sink for its set of urls
+    name = "pull-{0}".format(pull["number"])
+    for sink in SINKS:
+        for link in SINKS[sink]:
+            if link in ignore:
+                continue
+
+            # We only care about stuff at the sink where pull-XXXX is in
+            # the URL. This is how we figure out whether things are related
+            if name not in link:
+                continue
+
+            # Build a URL for the cockpituous sink /status file and read it
+            target = urlparse.urljoin(link, "status")
+            try:
+                data = json.loads(retrieve(target))
+            except ValueError, ex:
+                # print "JSON", ex
+                pass
+            except urllib2.HTTPError, ex:
+                if ex.code != 404:
+                    raise
+            except urllib2.URLError:
+                pass
+            else:
+                # The status file contains a "revision" field which is the git revision
+                # of what was tested during that test run. This is what we're after
+                if "revision" in data:
+                    if data["revision"] not in seen:
+                        seen[data["revision"]] = True
+                        yield data["revision"]
+
+# Pull out all status (context, created, log) for a given revision. This includes multiple
+# test runs for a given revision, and all the various status contexts
+def logs(revision):
+    page = 1
+    count = 100
+    while count == 100:
+        data = task.api.get("commits/{0}/status?page={1}&per_page={2}".format(revision, page, count))
+        for status in data.get("statuses", [ ]):
+            count += 1
+            if status["state"] not in [ "success", "failure" ]:
+                continue
+            target = status["target_url"]
+            if target.endswith(".html"):
+                target = target[:-5]
+            try:
+                yield (status["context"], status["created_at"], target, retrieve(target))
+            except urllib2.HTTPError, ex:
+                if ex.code != 404:
+                    raise
+            except urllib2.URLError:
+                pass
```

Comment:
Same logging question as above - we don't want to silently ignore real errors?

Reply:
Added logging